src/NLog/Internal/LogMessageTemplateFormatter.cs

```diff
@@ -35,70 +35,92 @@ namespace NLog.Internal
 {
     using System;
     using System.Globalization;
+    using System.Text;
     using MessageTemplates;
 
     internal sealed class LogMessageTemplateFormatter : ILogMessageFormatter
     {
-        public static readonly LogMessageTemplateFormatter DefaultAuto = new LogMessageTemplateFormatter(false);
-        public static readonly LogMessageTemplateFormatter Default = new LogMessageTemplateFormatter(true);
+        public static readonly LogMessageTemplateFormatter DefaultAuto = new LogMessageTemplateFormatter(false, false);
+        public static readonly LogMessageTemplateFormatter Default = new LogMessageTemplateFormatter(true, false);
+        public static readonly LogMessageTemplateFormatter DefaultAutoSingleTarget = new LogMessageTemplateFormatter(false, true);
+        public static readonly LogMessageTemplateFormatter DefaultSingleTarget = new LogMessageTemplateFormatter(true, true);
         private static readonly StringBuilderPool _builderPool = new StringBuilderPool(Environment.ProcessorCount * 2);
 
         /// <summary>
         /// When true: Do not fallback to StringBuilder.Format for positional templates
         /// </summary>
         private readonly bool _forceTemplateRenderer;
+        private readonly bool _singleTargetOnly;
 
         /// <summary>
         /// New formatter
         /// </summary>
         /// <param name="forceTemplateRenderer">When true: Do not fallback to StringBuilder.Format for positional templates</param>
-        private LogMessageTemplateFormatter(bool forceTemplateRenderer)
+        /// <param name="singleTargetOnly"></param>
+        private LogMessageTemplateFormatter(bool forceTemplateRenderer, bool singleTargetOnly)
         {
             _forceTemplateRenderer = forceTemplateRenderer;
+            _singleTargetOnly = singleTargetOnly;
             MessageFormatter = FormatMessage;
         }
 
         /// <summary>
         /// The MessageFormatter delegate
         /// </summary>
         public LogMessageFormatter MessageFormatter { get; }
-
+ 
         public bool HasProperties(LogEventInfo logEvent)
         {
-            //if message is empty, there no parameters
-            //null check cheapest, so in-front
+            if (HasParameters(logEvent))
+            {
+                if (_singleTargetOnly)
```

Comment:
micro: also _singleTargetOnly is cheaper to check than HasParameters?

Reply:
Will refactor the code, so it is more obvious what I'm trying to do.